src/gx/CGxDevice.hpp

```diff
@@ -120,6 +120,7 @@ class CGxDevice {
         virtual void PoolSizeSet(CGxPool*, uint32_t) = 0;
         virtual char* BufLock(CGxBuf*);
         virtual int32_t BufUnlock(CGxBuf*, uint32_t);
+        virtual void BufData(CGxBuf*, void*, size_t, uintptr_t);
```

Comment:
Let's make sure we name the parameters in the header declaration. I know that's duplicative, but it provides a reference for naming in the implementations, and IDEs generally show the header-declared parameter names as tooltips when developers are writing function calls.